### PR TITLE
Breaking: Remove preview instance prop from ContentPreview

### PR DESCRIPTION
**Pull request by DanDeMicco**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

File: src/components/ContentPreview/ContentPreview.js [hunk 1]
```diff
@@ -269,8 +269,10 @@ class ContentPreview extends PureComponent<Props, State> {
      *
      * @return {void}
      */
+    /* eslint-disable no-unused-vars */
     componentDidUpdate(prevProps: Props, prevState: State): void {
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: Here was the prior implementation without the need of componentDidUpdate / shouldLoadPreview
https://github.com/box/box-ui-elements/blob/00b9d1c624cab69644f96a6955cf9de87c9f3297/src/components/ContentPreview/ContentPreview.js#L378

Unsure if we can revert back to that?

Comment (priyajeet) [reply src/components/ContentPreview/ContentPreview.js]: Basically what happened before is that loadPreview() had a this.preview guard in it. In other words if this.preview was not undefined, it was assumed there was a preview open and no need to load preview.

Then only time this.preview was destroyed, when navigation happened.

Comment (priyajeet) [reply src/components/ContentPreview/ContentPreview.js]: I guess it was also destroyed in cWRP
https://github.com/box/box-ui-elements/blob/00b9d1c624cab69644f96a6955cf9de87c9f3297/src/components/ContentPreview/ContentPreview.js#L171

File: src/components/ContentPreview/ContentPreview.js [hunk 2]
```diff
@@ -270,8 +255,24 @@ class ContentPreview extends PureComponent<Props, State> {
...
+            this.destroyPreview();
+            this.fetchFile(fileId);
+        } else if (this.shouldLoadPreview(prevState)) {
             this.loadPreview();
+        } else if (hasTokenChanged) {
+            this.updatePreviewToken();
+        }
+    }
+
+    updatePreviewToken(shouldReload: boolean = false) {
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: Comments?

File: src/components/ContentPreview/ContentPreview.js [hunk 3]
```diff
@@ -270,8 +255,29 @@ class ContentPreview extends PureComponent<Props, State> {
...
+    }
+
+    /**
+     * Updates the access token used by preview library
+     *
+     * @param {boolean} shouldReload - true if preview should be reloaded
+     */
+    updatePreviewToken(shouldReload: boolean = false) {
+        if (this.preview) {
+            const { token } = this.props;
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: BTW, normally when only 1 prop is being used, and not repeated, its fine to use `this.props.token`

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| sidebar.js | 0 B| 0.00% | 997 KB | 997 KB |
| sidebar.css | 0 B| 0.00% | 79.7 KB | 79.7 KB |
| tree.js | 0 B| 0.00% | 632 KB | 632 KB |
| tree.css | 0 B| 0.00% | 22.3 KB | 22.3 KB |
| explorer.js | 0 B| 0.00% | 1.27 MB | 1.27 MB |
| explorer.css | 0 B| 0.00% | 96.9 KB | 96.9 KB |
| picker.js | 0 B| 0.00% | 715 KB | 715 KB |
| picker.css | 0 B| 0.00% | 41.9 KB | 41.9 KB |
| uploader.js | 0 B| 0.00% | 617 KB | 617 KB |
| uploader.css | 0 B| 0.00% | 31 KB | 31 KB |
| preview.js | 0 B| 0.00% | 1.04 MB | 1.04 MB |
| preview.css | 0 B| 0.00% | 81.4 KB | 81.4 KB |
| Total | 0 B | 0.00% | 5.55 MB | 5.55 MB |

Compared against previously merged PR #550

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

**Pull request merged**